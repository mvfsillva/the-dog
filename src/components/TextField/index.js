import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { transitions } from 'polished'

import { transition } from '@mixins/style.transtion'
import If from '@components/common/If'

const StyledInput = styled.input`
  background-color: transparent;
  border: 0;
  padding: 3px 0;
  width: 100%;
  margin: 5px;
  color: ${props => props.theme.colors.gray.default};
  border-bottom: 1px solid ${props => props.theme.colors.gray['300']};
  outline: snow;
  font-size: 1.5rem;
  text-transform: ${props => (props.lowercase ? 'lowercase' : 'capitalize')};

  &:hover {
    border-bottom-color: ${props => props.theme.colors.gray['500']};
  }

  &:focus {
    border-bottom-color: ${props => props.theme.colors.gray['500']};
  }

  &::placeholder {
    color: ${props => props.theme.colors.gray['600']};
  }

  ${transitions(transition({ property: 'border-bottom-color' }))};
`

const StyledLabel = styled.label`
  color: ${props => props.theme.colors.gray.mercury};
  font-size: 1.4rem;
  letter-spacing: 5px;
  line-height: 40px;
  margin: 5px;
  font-weight: 300;

  input:focus + & {
    color: ${props => props.theme.colors.gray['500']};
  }

  input:hover + & {
    color: ${props => props.theme.colors.gray['500']};
  }

  ${transitions(transition({ property: 'color' }))};
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: column-reverse;
  margin-bottom: 1.2rem;
  label {
    vertical-align: middle;
  }
`

const TextField = ({ name, label, type, ...props }) => {
  return (
    <Wrapper>
      <StyledInput id={name} name={name} type={type} {...props} />
      <If test={label}>
        <StyledLabel htmlFor={name}>{label}</StyledLabel>
      </If>
    </Wrapper>
  )
}

TextField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.string.isRequired,
  lowercase: PropTypes.bool,
}

export default TextField
