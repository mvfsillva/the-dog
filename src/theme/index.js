const baselineGrid = 4
const interfaceGrid = 8

const breakpoints = ['576px', '768px', '992px', '1200px']

const colors = {
  primary: {
    default: '#6800c3',
    fallback: '#a55cff',
    text: '#fff',
  },
}

export default {
  baselineGrid,
  interfaceGrid,
  breakpoints,
  colors,
}
