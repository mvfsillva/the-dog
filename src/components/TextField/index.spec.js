import React from 'react'
import 'jest-styled-components'
import theme from '../../theme'
import { shallowWithTheme, mountWithTheme } from '../../../config/test/enzyme.theme'
import TextField from '.'

const shallow = shallowWithTheme(theme)
const mount = mountWithTheme(theme)

const wrap = (props = {}) => shallow(<TextField type="text" name="name" label="label" {...props} />)

describe('[Component: TextField]', () => {
  it('mounts with different combination props', () => {
    mount(<TextField type="text" name="name" label="label" />)
    mount(<TextField type="email" name="name" label="label" lowercase />)
  })
  it('renders input props when passed in', () => {
    const wrapper = wrap({ id: 'foo' })
    expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
  })

  it('renders label when passed in', () => {
    const wrapper = wrap({ name: 'name' })
    expect(wrapper.find({ name: 'name' })).toHaveLength(1)
  })

  it('renders label when passed in', () => {
    const wrapper = wrap({ label: 'foo label' })
    expect(wrapper.contains('foo label')).toBe(true)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<TextField>TextField component</TextField>)
    expect(wrapper).toMatchSnapshot()
  })
})
