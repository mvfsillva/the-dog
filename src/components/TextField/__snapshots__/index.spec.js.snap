// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: TextField] matches snapshot 1`] = `
<styled.div>
  <styled.input
    invalid={false}
  >
    TextField component
  </styled.input>
  <If>
    <styled.label />
  </If>
</styled.div>
`;
